### ISSUE TITLE
GDAL fails to build on High Sierra with CLTools installed

### ISSUE THREAD

**Issue by jerr0328** [OSGeo/homebrew-osgeo4mac#276]
Just noting that this is a known issue that I've been running into: https://github.com/Homebrew/homebrew-core/issues/14418#issuecomment-324257556

For instance:
```In file included from cpl_conv.cpp:54:
In file included from /private/tmp/gdal2-20170927-17481-1shd4hy/gdal-2.2.2/port/cpl_conv.h:35:
In file included from /private/tmp/gdal2-20170927-17481-1shd4hy/gdal-2.2.2/port/cpl_port.h:659:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cmath:313:9: error: no member named 'signbit' in the global namespace
using ::signbit;
      ~~^```

**Comment by tomcardoso** [comment]
Happening for me too on (non-High) Sierra, FWIW. Can I uninstall Xcode or something? Would that work?

**Comment by dakcarto** [comment]
@tomcardoso, did you upgrade to Xcode 9? If so, you could revert to Xcode 8.3.3 or 8.2.1. If you have use the App Store to install, it may be tricky to revert. Usually you need to download older versions directly from Apple, though that requires a Developer account (free now):

https://developer.apple.com/download/more/

Of course, install the CLTools as well for your macOS version.

**Comment by russmo** [comment]
Yes, @tomcardoso, I just did that a few minutes ago to install gdal2.  I dragged the Xcode app from the /Applications folder to the trash.  I made sure there wasn't anything at /Library/Developer/CommandLineTools.  Then installed the command line tools for Xtools 8.3.3 rather than Xtools 9. You can download past versions here https://developer.apple.com/download/more/

**Comment by jerr0328** [comment]
Workaround is to force the use of the Sierra bottle with `brew install --force-bottle gdal2` or `brew upgrade --force-bottle gdal2`.

**Comment by dakcarto** [comment]
@jerr0328 wrote:
> Workaround is to force the use of the Sierra bottle with brew install --force-bottle gdal2 or brew upgrade --force-bottle gdal2.

I thought about just duplicating the bottle for Sierra over to High Sierra and updating the bottle block in the formula, but didn't yet have a VM to test it in. While this is not ideal, I'm guessing it is an OK workaround for now. Thanks for testing that.

**Comment by dakcarto** [comment]
Anyone on High Sierra:

What's the bottle archive file name syntax? For Sierra it is `gdal2-2.2.2.sierra.bottle.tar.gz`. Is it `gdal2-2.2.2.highsierra.bottle.tar.gz` or `gdal2-2.2.2.high-sierra.bottle.tar.gz`?

**Comment by jerr0328** [comment]
@dakcarto neither, it's `high_sierra` ;)

**Comment by dakcarto** [comment]
@jerr0328, Sierra bottle duplicated in 5335a48. See if that works, when you get a chance. Thanks.

**Comment by jerr0328** [comment]
Yeah this works for me.

**Comment by dakcarto** [comment]
**Update:** I think we basically have to wait until the Xcode 9.0 CLTs are updated. I haven't found a workaround so far.

**Comment by fbonzon** [comment]
@dakcarto Command Line Tools for Xcode 9.0.1 were released yesterday, but I still see the compilation error.

**Comment by dakcarto** [comment]
@fbonzon, thanks for the update.

Anyone... does this work ([culled from here](https://github.com/davidchall/homebrew-hep/issues/108#issuecomment-336341177))?
```ruby
ENV.delete("SDKROOT") if DevelopmentTools.clang_build_version >= 900
```

Add right before `system "./configure", *configure_args` for `gdal2` main build.

(Sorry, don't have time right now to test.)

**Comment by nickrobison** [comment]
That fix worked for me on 10.13 with Xcode 9.0.1.

**Comment by dakcarto** [comment]
@nickrobison, thanks for the feedback! I still haven't tested, but will push the change as a temporary fix.
